saltgui/static/scripts/routes/job.js

```diff
@@ -20,7 +20,7 @@ class JobRoute extends Route {
     job.getPageElement().querySelector(".output").innerText = "";
 
     document.querySelector("#button_close_job").addEventListener("click", _ => {
-      this.router.goTo("/");
+      this.router.goTo("/jobs");
```

Comment:
Can this be re-organized to go back to the page where the user came from?
One can end up on the jobs page directly from the home page, so you want to return to the home page in that case.

Reply:
Thx for the hint. I've ended up with this: `window.location.reload(history.back());`

Reply:
Unfortunately, that does not work...
(also: `history.back` does not have a return value and `location.reload` does not take a parameter. it should have been 2 separate commands, but that does not change the situation)
The current navigation framework uses a mix of `location.assign` and `location.replace`, combined with explicit `history.pushState` manipulations. Let's not mess with that now. The current version already has the same problem when showing job details from most pages.
Let's return to your original code of using `/jobs'. Sorry for the detour.

Reply:
Ok. I will revert this part - strange as this was really working in my case (firefox browser).

Reply:
I had the same conclusion about the combined code and also used the newly suggested code. Now retried it again, but no success.
It works on all pages, except for job details reached from the jobs page itself.

Reply:
SaltGUI is mostly using `location.replace`, thus destroying the history. See also https://www.w3schools.com/jsref/met_loc_replace.asp. Let's not rely on `windows.history`.

Reply:
I found out why it was not working on jobs page. On most pages we are going directly from particular page to job details (so `history.go(-1)` is enough) however on jobs page we are going through event callback (so `history.go(-1)` is not enough and it will work only with `history.go(-2)` ).

So I was able to workaround this by this code:
```
    document.querySelector("#button_close_job").addEventListener("click", _ => {
      window.history.back();
      // when we were in jobs page we need to bo back twice
      if (window.location.pathname == '/job') {
        window.history.back();
      } else {
        // on rest of the pages we need to reload page
        window.location.reload();
      }
    });
```

Reply:
Anyway code reverted to `this.router.goTo("/jobs");`